common/utilities/string/string-utilities.h

```diff
@@ -3,139 +3,189 @@
-            }
-            else if( std::is_same< T, unsigned long long >::value )
-            {
-                if( str[0] == '-' )
-                    return false;
-                result = static_cast< T >( std::stoull( str, &last_char_idx ) );
-            }
-            else if( std::is_same< T, long >::value )
-            {
-                result = static_cast< T >( std::stol( str, &last_char_idx ) );
-            }
-            else if( std::is_same< T, long long >::value )
+            std::string res;
+
+            do
             {
-                result = static_cast< T >( std::stoll( str, &last_char_idx ) );
-            }
-            else if( std::is_same< T, int >::value )
+                res += "0123456789ABCDEF"[n % 16];
+                n >>= 4;
+            } while (n);
+
+            std::reverse(res.begin(), res.end());
+
+            if (res.size() == 1)
             {
-                result = static_cast< T >( std::stoi( str, &last_char_idx ) );
+                res.insert(0, "0");
             }
-            else if( std::is_same< T, short >::value )
-            {  // no dedicated function fot short in std
 
-                int check_value = std::stoi( str, &last_char_idx );
-
-                if( check_value > std::numeric_limits< short >::max()
-                    || check_value < std::numeric_limits< short >::min() )
-                    throw std::out_of_range( "short" );
+            return res;
+        }
 
-                result = static_cast< T >( check_value );
-            }
-            else if( std::is_same< T, unsigned int >::value )
-            {  // no dedicated function in std - unsgined corresponds to 16 bit, and unsigned long
-               // corresponds to 32 bit
-                if( str[0] == '-' )
-                    return false;
+        inline std::string to_lower(std::string x)
+        {
+            std::transform(x.begin(), x.end(), x.begin(), tolower);
+            return x;
+        }
 
-                unsigned long check_value = std::stoul( str, &last_char_idx );
+        inline std::string to_upper(std::string x)
+        {
+            std::transform(x.begin(), x.end(), x.begin(), toupper);
+            return x;
+        }
 
-                if( check_value > std::numeric_limits< unsigned int >::max() )
-                    throw std::out_of_range( "unsigned int" );
+        inline bool string_to_bool(const std::string& x)
+        {
+            return (to_lower(x) == "true");
+        }
 
-                result = static_cast< T >( check_value );
-            }
-            else if( std::is_same< T, unsigned short >::value
-                     || std::is_same< T, unsigned short >::value )
-            {  // no dedicated function in std - unsgined corresponds to 16 bit, and unsigned long
-               // corresponds to 32 bit
-                if( str[0] == '-' )
-                    return false;
+        inline unsigned int string_to_hex(std::string str)
```

Comment:
This is more like `hex_to_uint`. Can even make is a template, but it's using all kind of extra classes like `stringstream` which really aren't needed... why not just use `sscanf`? If you insist on using stringstream, at least `#include <sstream>`.

`delimiter` isn't needed (especially not as a string).

Reply:
The func converts ASCII representation of hex value ('0xAB3F01FA' ) back to UINT format

Reply:
Right... hence `hex_to_uint`
By the way, please pass `str` by `const &`

Reply:
The call does in-place modifications of the variable, thus passed by value